src/services/publisher_service/helpers.rs

```diff
@@ -173,3 +173,43 @@ pub async fn update_message_processing_status(
         .await?;
     Ok(())
 }
+
+/// Checks for messages in the `processing` state for more than threshold in minutes
+/// and put it back in a `queued` state for processing
+#[instrument(skip(postgres))]
+pub async fn dead_letters_check(
+    threshold_minutes: i8,
+    postgres: &PgPool,
+) -> std::result::Result<(), sqlx::error::Error> {
+    let update_status_query = "
+        UPDATE subscriber_notification
+        SET status = 'queued'
+        WHERE status = 'processing'
+        AND EXTRACT(EPOCH FROM (NOW() - updated_at))/60 > $1::INTEGER
+    ";
+    sqlx::query::<Postgres>(update_status_query)
+        .bind(threshold_minutes)
+        .execute(postgres)
+        .await?;
+    Ok(())
+}
+
+/// Checks for message is created more than threshold in minutes
+#[instrument(skip(postgres))]
+pub async fn dead_letter_give_up_check(
+    notification: Uuid,
+    threshold_minutes: i16,
```

Comment:
These threshold parameters and the constants should be `Duration`s

Reply:
I think it doesn't look like makes sense here, because of:
* Duration maximum constructor and getter is for seconds, but we operate with minutes and above up to days. So in seconds, the value will be larger, and the operation to convert Duration into seconds doesn't make sense as the variable is self-describing here and doesn't need any later convertions.
* Duration should still be converted to minutes because we are using minutes values in SQL queries where the calculation is for minutes. We can create a duration from seconds and then calculate minutes for the SQL query, for example, but what is the purpose of that additional overhead?
* We can use only seconds for the threshold and get it as `as_secs` later and use seconds instead of minutes in SQL queries, but we will have an additional conversion and a bigger allocation size instead of just using the i16 small type.
So it looks like an additional overhead and possibly overengineering here to me 🤷‍♂️

Reply:
The SQL query be in seconds instead of minutes. Typically things are usually represented in seconds or milliseconds anyway?

While the variable name is self-describing, it's more readable and safer to work with a duration so we don't accidentally switch units in plumbing code, or call the function a different time with the wrong unit. Not sure 6 byte memory savings is worth anything compared to the size of the SQL statement and logs and other stuff. I like reading abstract Durations in my code instead of having to think about numbers or minutes